src/txn/txn_ckpt.c

```diff
@@ -1426,6 +1419,18 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
      */
     if (full || !logging) {
         session->isolation = txn->isolation = WT_ISO_READ_UNCOMMITTED;
+
+        /*
+         * Checkpoint the shared metadata table last, as it could have changed. Checkpoint it after
+         * we have released the checkpoint transaction. Otherwise, we may evict it and the
+         * checkpoint transaction cannot commit in this case as the updates have gone from memory.
```

Comment:
Could you clarify this comment - I don't understand the justification.

Reply:
Checkpoint writes to these tables. With its own snapshot, it can write and clean these pages even the updates are not committed. If the page is clean, they can be evicted and we hit the memory corruption issue when we commit the checkpoint transaction.

Reply:
[0022d36](https://github.com/wiredtiger/wiredtiger/pull/11438/commits/0022d369c0972d37d7427cd0f84d4228a068f6df)